gramineproject/gramine
# [tools] Split `libsgx_util.a` in IAS lib and other-utils lib

## Description of the changes 

Previously, Gramine's `libsgx_util.a` contained both the IAS (Intel Attestation Service, used for EPID SGX remote attestation) functionality and the other functionality (protected files, string/debug utilities, SGX quote utilities). This mix was wrong because this static lib is installed together with Gramine, but IAS part is not stand-alone -- it depends on `libcurl.a`, which is not installed together with Gramine.

Meson v1.2.0 began to fail on this "installed static lib that links against a not-installed static lib" scenario (previously Meson only printed a warning which we overlooked). This PR splits the functionality of `libsgx_util.a` into two parts: one Gramine-internal lib `libsgx_ias_util.a` used only for IAS binaries, and the other `libsgx_util.a` used for everything else (including used by external users).

This PR doesn't introduce any regression, since external users were never supposed to use IAS-specific APIs.

Fixes #1459.

See #1459 for the context and explanations.

## How to test this PR? 

CI tests that the new split doesn't break our existing tools.

I also manually verified tools like `gramine-sgx-ias-verify-report`.

Finally, one can verify with GSC that doesn't have the `!=1.2.*` trick and uses this branch.


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gramineproject/gramine/1472)
